#!/usr/bin/env python
PACKAGE = "nostop_kinect_sensor"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("R_distance", double_t, 0, "From camera point on the floor to center of view", 150,  1, 500)

gen.add("W_xC", double_t, 0, "x camera coordinate in world frame", 0,  -500, 500)

gen.add("W_yC", double_t, 0, "y camera coordinate in world frame", 0,  -500, 500)

gen.add("W_zC", double_t, 0, "z camera coordinate in world frame", 100,  1, 300)

gen.add("W_xAxesP", double_t, 0, "x coordinate of P point to found z rotation of C to go to W", 0,  -500, 500)

gen.add("W_yAxesP", double_t, 0, "y coordinate of P point to found z rotation of C to go to W", 0,  -500, 500)

# gen.add("omega_zC", double_t, 0, "rotation around zCamera to have xCamera in xWorld", 0, -180, 180)

gen.add("gamma_xC", double_t, 0, "rotation around xCamera to have yCamera in yWorld ( DO NOT CHANGE!!! )", 180, -180, 180)

gen.add("h_robot", double_t, 0, "height of robots", 3.5, 0, 50)

exit(gen.generate(PACKAGE, "nostop_kinect_sensor", "Camera_calibration"))
